tests/functional/object/rgw/test_rgw_sts.py

```
@@ -226,11 +265,138 @@ def get_object(self, s3_client, bucket_name, object_key):
+        Assert that a boto3 operation raises a ClientError with the expected error code.
+
+        Args:
+            operation_fn (callable): Zero-argument callable that performs the operation
+            expected_error_code (str): Expected error code (e.g. "AccessDenied")
+            step_label (str): Label used in log messages (e.g. "Step 5")
+        """
+        try:
+            operation_fn()
+            assert (
+                False
+            ), f"{step_label}: Expected {expected_error_code} but operation succeeded"
```

Ruff (0.15.20)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)